tests/lib/utilities.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+wait_for_systemd_service() {
+	while ! systemctl status $1 ; do
+		sleep 1
+	done
+	sleep 1
+}
+
+wait_for_daemon_ready() {
+	wait_for_systemd_service snap.wifi-connect.daemon.service
+}
+
+wait_for_systemd_service_exit() {
+	count=20
+	while systemctl status $1 && count -gt 0; do
+		sleep 1
+		let count--
+	done
+	sleep 1
+
+	if [ count -eq 0 ]; then
+		exit 1
+	fi
+}
+
+does_interface_exist() {
+	[ -d /sys/class/net/$1 ]
+}
+
+wait_until_interface_is_available() {
+	while ! does_interface_exist $1; do
+		# Wait for 200ms
+		sleep 0.2
+	done
+}
+
+stop_after_first_reboot() {
+	if [ $SPREAD_REBOOT -gt 0 ] ; then
+		exit 0
+	fi
+}
+
+install_snap() {
+	# Don't reinstall if we have it installed already
+	if ! snap list | grep $1 ; then
+		snap install --$2 $1
+	fi
+}
+
+install_additional_snaps() {
+	install_snap network-manager stable
```

Comment:
added a spread var (ADDITIONAL_SNAPS_CHANNEL) to modify as needed. Setting stable as value in that var